hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndex.java

```diff
@@ -97,13 +98,20 @@ public abstract O updateLocation(O writeStatuses, HoodieEngineContext context,
   @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
   public abstract boolean isImplicitWithStorage();
 
+  /**
+   * Get the index Type.
+   */
+  public IndexType indexType() {
```

Comment:
should we leave this abstract? So, any new index is forced to implement this? Downside is, when people upgrade, they may need to implement the method (may be ok). 
I don't think many outside users have custom indexes now. cc @leesf do you know of any?

Reply:
okay